### PR TITLE
flow: postpone FlowForceReassembly at shutdown

### PR DESCRIPTION
**Pull request by fooinha**
- fixes Bug 1946.

More information at redmine. Close it if non-sense.

### CODE REVIEW

File: src/suricata.c [hunk 1]
```diff
@@ -2698,6 +2692,12 @@ int main(int argc, char **argv)
         FlowDisableFlowRecyclerThread();
     }
 
+    if (suri.run_mode != RUNMODE_UNIX_SOCKET) {
+        /* we need a packet pool for FlowForceReassembly */
+        PacketPoolInit();
+        FlowForceReassembly();
```
Comment (inliniac) [inline src/suricata.c]: Whats the logic of moving this to after the packet processing threads are killed? How are timeout pseudo packets processed?

Comment (fooinha) [inline src/suricata.c]: I don't know enough to answer that.

What I saw, was that the FlowForceReassembly() was called while processing flows, and after breaking main's loop , and as a side effect, the http output was borked.

For instance, by adding a single time(1) right after suri's main loop, the http output was complete and ok for the supplied pcap files in redmine ticket 1946.

**Comment by inliniac** [comment]
@fooinha  I'm looking at this, but can't seem to reproduce it. Could you help me out by sharing the test method and script? Thanks!

**Comment by fooinha** [comment]
Fixed by #2518

**Pull request closed without merging**